### PR TITLE
Add snowflake proto

### PR DESCRIPTION
**Pull request by pingsutw**
Signed-off-by: Kevin Su <pingsutw@apache.org>

# TL;DR
Add snowflake in FLyteidl for the snowflake plugin to use. 
We will get "query statement"  from the target instead of custom, so I didn't define "statement" in his proto.
Related to https://github.com/flyteorg/flyteidl/pull/214

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [x] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/704
https://github.com/flyteorg/flyteplugins/pull/202

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by EngHabu** [comment]
Closing in favor of: https://github.com/flyteorg/flyteidl/pull/214
